packages/experimental/Button/src/useButton.ts

```diff
@@ -8,7 +8,7 @@ export const useButton = (props: ButtonPropsWithInnerRef): ButtonState => {
   const defaultRef = React.useRef(null);
   const { onClick, innerRef, disabled, loading, ...rest } = props;
   const ref = innerRef !== null ? innerRef : defaultRef;
-  const onClickWithFocus = useOnPressWithFocus(ref, onClick);
+  const onClickWithFocus = disabled ? null : useOnPressWithFocus(ref, onClick);
```

Comment:
This feels like it violates one of the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level): Don't call hooks conditionally. I would rather we unconditionally call the hook, pass in `disabled` as a prop to the hook, and only focus if `!disabled`. I also still feel like this is a workaround and might be worth leaving a comment in code explaining why we need it / the bug # to track why just setting `focusable` isn't enough

Reply:
Good point, we shouldn't call hooks conditionally to ensure the order of execution is the same in every component render.

How about we pass in null as `ref` if button is disabled so that the hook bypasses calling .focus()? (I think this is overriding `focusable` which explains why the disabled button gets keyboard focus) I prefer this approach because there's seems to be assumption that the component is pressable when `useOnPressWithFocus` is called.

Reply:
That could work, though it still feels like a workaround I would leave a comment / bug number with. What do you mean `focusable` is overridden? The hook doesn't seem to touch how we define `focusable` (line 25).

Reply:
I meant to say that the setter for focusable is disregarded because we're calling .focus() in `useOnPressWithFocus`  if `ref` is valid.

Reply:
IMO, calling `ref.focus()` should not focus on the view if `focusable: false`. That feels like the real bug to me.
Natively on macOS, it seems we are also checking for focusable when we call `acceptsFirstResponder` too https://github.com/microsoft/react-native-macos/blob/877d09aa862aea049cd58ec1545153c7574fc662/React/Views/RCTView.m#L1407

Reply:
Yeah that's weird. `useOnPressWithFocus` respected the disabled prop with no issue on win32. That workaround seems good for now though.

Reply:
Thank you for the feedback Saad and Krystal, I updated the [GH issue](https://github.com/microsoft/fluentui-react-native/issues/1336) to better explain the bug.